src/scripts/main.js

```diff
@@ -1,3 +1,301 @@
+            )
+          );
+        };
+        break;
+
+      default:
+        compare = function (rowA, rowB) {
+          return rowA.cells[index].innerHTML > rowB.cells[index].innerHTML
+            ? 1
+            : -1;
+        };
+    }
+    arrayOf.sort(compare);
+    lastClickedIndex = index;
+  } else {
+    arrayOf.reverse();
+  }
+
+  arrayOf.forEach((tr) => {
+    tBody.appendChild(tr);
+  });
+}
+
+tBody.addEventListener('click', (e) => {
+  const focusedRow = e.target.closest('TR');
+
+  if (!focusedRow || activeRow === focusedRow) {
+    return;
+  }
+
+  if (activeRow) {
+    activeRow.classList.remove('active');
+  }
+  activeRow = focusedRow;
+  activeRow.classList.add('active');
+});
+
+function validateForm(formData) {
+  if (
+    !formData.name ||
+    !formData.position ||
+    !formData.age ||
+    !formData.salary
+  ) {
+    pushNotification('Warning!', 'Each field must be filled in!', 'error');
+
+    return false;
+  }
+
+  if (typeof formData.name !== 'string') {
+    pushNotification('Warning!', 'Name should be a string!', 'warning');
+
+    return false;
+  }
+
+  if (typeof formData.position !== 'string') {
+    pushNotification('Warning!', 'Position should be a string!', 'warning');
+
+    return false;
+  }
+
+  if (names.includes(formData.name)) {
+    pushNotification('Warning!', 'This name already exist!', 'warning');
+
+    return false;
+  }
+
+  if (formData.name.length < 4) {
+    pushNotification('Error!', 'Name must not to be shorter than 4!', 'error');
+
+    return false;
+  }
+
+  if (formData.age < minAge || formData.age > maxAge) {
+    pushNotification(
+      'Error!',
+      `Age have to be between ${minAge} and ${maxAge}!`,
+      'error',
+    );
+
+    return false;
+  }
+
+  if (formData.position.length < 5) {
+    pushNotification(
+      'Warning!',
+      'Position field have to have more then 5 characters!',
+      'error',
+    );
+
+    return false;
+  }
+
+  return true;
+}
+
+function addRow(data) {
+  const newRow = document.createElement('tr');
+
+  Object.entries(data).forEach(([key, value]) => {
+    const newCell = document.createElement('td');
+
+    newCell.textContent =
+      key === 'salary' ? `$${value.toLocaleString()}` : value;
+
+    newRow.appendChild(newCell);
+  });
+
+  tBody.appendChild(newRow);
+}
+
+tBody.addEventListener('dblclick', (e) => {
+  const dblclickedCell = e.target.closest('td');
+  const lastValue = dblclickedCell.textContent;
+
+  dblclickedCell.textContent = '';
+
+  const cellInput = document.createElement('input');
+
+  cellInput.classList.add('cell-input');
+  dblclickedCell.appendChild(cellInput);
+  cellInput.focus();
+
+  cellInput.addEventListener('blur', (ee) => {
+    dblclickedCell.textContent = lastValue;
+  });
+
+  cellInput.addEventListener('keypress', (inputEvent) => {
+    if (inputEvent.key === 'Enter') {
+      /* dblclickedCell.textContent = ''; */
+      dblclickedCell.textContent = cellInput.textContent;
```

Comment:
The line `dblclickedCell.textContent = cellInput.textContent;` is incorrect. You should use `cellInput.value` instead of `cellInput.textContent` to get the value from the input field.